contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,428 @@
+
+  private Schema schema;
+
+  private List<FieldInfo> fieldInfos;
+
+  private transient List<ActiveFieldInfo> columnFieldSetters = null;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  @AutoMetric
+  int fieldErrorCnt = 0;
+
+  public void parseSchema() throws IOException
+  {
+    setSchema(new Schema.Parser().parse(getSchemaString()));
+  }
+
+  public String getSchemaString()
+  {
+    return schemaFile;
+  }
+
+  public void setSchemaString(String schemaFile)
+  {
+    this.schemaFile = schemaFile;
+  }
+
+  public Schema getSchema()
+  {
+    return schema;
+  }
+
+  public void setSchema(Schema schema)
+  {
+    this.schema = schema;
+  }
+
+  public byte[] serialize(Object obj) throws IOException
+  {
+    try (ByteArrayOutputStream b = new ByteArrayOutputStream()) {
+      try (ObjectOutputStream o = new ObjectOutputStream(b)) {
+        o.writeObject(obj);
+      }
+      return b.toByteArray();
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = false)
+  public final transient DefaultInputPort<GenericRecord> data = new DefaultInputPort<GenericRecord>()
+  {
+
+    @Override
+    public void process(GenericRecord tuple)
+    {
+
+      try {
+        Object obj = getPOJOFromGenericRecord(tuple, getCls());
+
+        if (obj != null) {
+          output.emit(obj);
+          recordCnt++;
+        }
+
+      } catch (InstantiationException | IllegalAccessException e) {
+        LOG.error("Could not initialize object of class - " + getClass().getName());
+        errorCnt++;
+      }
+    }
+
+  };
+
+  @SuppressWarnings("unchecked")
+  public Object getPOJOFromGenericRecord(GenericRecord tuple, Class<?> cls)
+      throws InstantiationException, IllegalAccessException
+  {
+    Object newObj = getCls().newInstance();
+
+    try {
+
+      for (int i = 0; i < columnFieldSetters.size(); i++) {
+
+        AvroToPojo.ActiveFieldInfo afi = columnFieldSetters.get(i);
+
+        afi = columnFieldSetters.get(i);
+
+        SupportType st = afi.fieldInfo.getType();
+
+        Object val = null;
+
+        try {
+          val = tuple.get(afi.fieldInfo.getColumnName());
+        } catch (Exception e) {
+          LOG.error("Could not find field -" + afi.fieldInfo.getColumnName() + "- in the generic record");
+          val = null;
+          fieldErrorCnt++;
+        }
+
+        if (val == null) {
+          continue;
+        }
+
+        try {
+
+          switch (st) {
+
+            case BOOLEAN:
+
+              afi.setterOrGetter = PojoUtils.createSetterBoolean(getCls(), afi.fieldInfo.getPojoFieldExpression());
+              ((PojoUtils.SetterBoolean<Object>)afi.setterOrGetter).set(newObj,
+                  (boolean)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case DOUBLE:
+              afi.setterOrGetter = PojoUtils.createSetterDouble(getCls(), afi.fieldInfo.getPojoFieldExpression());
+              ((PojoUtils.SetterDouble<Object>)afi.setterOrGetter).set(newObj,
```

Comment:
I'm not sure if this would work if the output POJO has wrapper classes.
Can you please add such a test?